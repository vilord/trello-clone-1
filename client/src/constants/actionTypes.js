/**
 * USER
 */
export const SIGNUP_USER_REQUEST = 'SIGNUP_USER_REQUEST';
export const SIGNUP_USER_SUCCESS = 'SIGNUP_USER_SUCCESS';
export const SIGNUP_USER_FAILURE = 'SIGNUP_USER_FAILURE';

export const LOGIN_USER_REQUEST = 'LOGIN_USER_REQUEST';
export const LOGIN_USER_SUCCESS = 'LOGIN_USER_SUCCESS';
export const LOGIN_USER_FAILURE = 'LOGIN_USER_FAILURE';

export const LOGOUT_USER = 'LOGOUT_USER';

export const GET_USER_SESSION_REQUEST = 'GET_USER_SESSION_REQUEST';
export const GET_USER_SESSION_SUCCESS = 'GET_USER_SESSION_SUCCESS';

export const SET_USER_PROFILE_REQUEST = 'SET_USER_PROFILE_REQUEST';
export const SET_USER_PROFILE_SUCCESS = 'SET_USER_PROFILE_SUCCESS';
export const SET_USER_PROFILE_FAILURE = 'SET_USER_PROFILE_FAILURE';

/**
 * UI
 */
export const SET_UI_ERROR = 'SET_UI_ERROR';
export const RESET_UI_ERROR = 'RESET_UI_ERROR';

export const SHOW_BOARDS_EXPLORER = 'SHOW_BOARDS_EXPLORER';
export const HIDE_BOARDS_EXPLORER = 'HIDE_BOARDS_EXPLORER';

export const FOCUS_HEADER_SEARCH = 'FOCUS_HEADER_SEARCH';
export const BLUR_HEADER_SEARCH = 'BLUR_HEADER_SEARCH';

export const SHOW_CREATE_MENU = 'SHOW_CREATE_MENU';
export const HIDE_CREATE_MENU = 'HIDE_CREATE_MENU';

export const SHOW_NOTIFICATIONS = 'SHOW_NOTIFICATIONS';
export const HIDE_NOTIFICATIONS = 'HIDE_NOTIFICATIONS';

export const SHOW_USER_MENU = 'SHOW_USER_MENU';
export const HIDE_USER_MENU = 'HIDE_USER_MENU';

export const SHOW_BOARD_MENU = 'SHOW_BOARD_MENU';
export const HIDE_BOARD_MENU = 'HIDE_BOARD_MENU';

/**
 * Cards
 */
export const SHOW_CARD_DETAIL = 'SHOW_CARD_DETAIL ';
export const HIDE_CARD_DETAIL = 'HIDE_CARD_DETAIL ';

export const SHOW_LABEL_TEXT = 'SHOW_LABEL_TEXT';
export const HIDE_LABEL_TEXT = 'HIDE_LABEL_TEXT';
