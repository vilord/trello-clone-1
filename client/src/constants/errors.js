export const invalidEmail = {
  kind: 'invalidEmail',
  header: 'Verify email!',
  message: "That doesn't look like an email addressâ€¦",
};

export const shortPassword = {
  kind: 'shortPassword',
  header: 'Password is too short!',
  message: 'Your password must be at least 8 characters long.',
};

export const userExists = {
  kind: 'userExists',
  header: 'User already exists!',
  message: 'Email is already in use by another account.'
};

export const wrongCredentials ={
  kind: 'wrongCredentials',
  header: 'Something wrong!',
  message: 'Incorrect email (username) or password.'
};

export const serverError = {
  kind: 'serverError',
  header: 'Unknown Server Error!',
  message: 'Our server may be down. Please retry later!'
};
