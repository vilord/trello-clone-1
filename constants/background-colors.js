module.exports = {
  BLUE: '#127ABD',
  ORANGE: '#D18F3E',
  GREEN: '#53973F',
  RED: '#AE4736',
  PURPLE: '#88629C',
  PINK: '#CB5C91',
  LIGHT_GREEN: '#50BE6E',
  LIGHT_BLUE: '#1CAECA',
  GREY: '#838C91',
};
