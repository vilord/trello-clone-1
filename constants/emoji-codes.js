module.exports = [
  'thumbsup',
  'thumbsdown',
  'raised_hands',
  'pray',
  'eyes',
  'rocket',
  'joy',
  'blush',
  'hearth_eyes',
  'throphy',
  'heart',
  'broken_heart',
];
