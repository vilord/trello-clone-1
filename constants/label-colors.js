module.exports = {
  GREEN: '#64BB55',
  YELLOW: '#F5DF63',
  ORANGE: '#FDAA54',
  RED: '#E95B4B',
  PURPLE: '#C27ADE',
  BLUE: '#127ABD',
  LIGHT_BLUE: '#20C2DE',
  LIGHT_GREEN: '#59E79A',
  PINK: '#FD83CD',
  DARK_GREY: '#4D4D4D',
};
