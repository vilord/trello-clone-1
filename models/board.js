const mongoose = require('mongoose');
const isURL = require('validator/lib/isURL');

const Schema = mongoose.Schema;
const { ObjectId } = Schema.Types;

// Constants
const backgroundColors = require('../constants/background-colors');
const { BLUE } = backgroundColors;

const Board = new Schema({
  title: {
    type: String,
    required: true,
  },
  visibility: {
    type: String,
    enum: ['public', 'team', 'private'],
    default: 'public',
  },
  lists: [
    {
      type: ObjectId,
      ref: 'List',
    },
  ],
  labels: [
    {
      type: ObjectId,
      ref: 'Label',
    },
  ],
  members: [
    {
      user: {
        type: ObjectId,
        ref: 'User',
      },
      admin: {
        type: Boolean,
        default: false,
      },
    },
  ],
  activity: [
    {
      type: ObjectId,
      ref: 'Activity',
    },
  ],
  theme: {
    color: {
      type: String,
      enum: Object.values(backgroundColors),
      default: BLUE,
    },
    picture: {
      type: String,
      validate: {
        validator: x => isURL(x),
        message: 'Must be a valid URL',
      },
    },
  },
});

module.exports = mongoose.model('Board', Board);
