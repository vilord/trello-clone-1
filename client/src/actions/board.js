import * as types from '../constants/actionTypes';

export const setBoardTitle = title => ({
  type: types.SET_BOARD_TITLE,
  title,
});
