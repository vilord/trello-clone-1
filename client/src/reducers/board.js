import * as types from '../constants/actionTypes';
import initState from './initBoardState';

const boardReducer = (state = initState, action) => {
  switch (action.type) {
    case types.GET_BOARD_SUCESS:
      return {
        ...state,
        ...action.board,
      };
    default:
      return state;
  }
};

export default boardReducer;
