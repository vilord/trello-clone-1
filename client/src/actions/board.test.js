import * as actions from './board';
import * as types from '../constants/actionTypes';

describe('Board Actions', () => {
  describe('Action Creators', () => {
    it('creates a SET_BOARD_TITLE action', () => {
      const expected = {
        type: types.SET_BOARD_TITLE,
        title: 'Some Title',
      };

      expect(actions.setBoardTitle(expected.title)).toEqual(expected);
      expect(actions.setBoardTitle(expected.title)).toMatchSnapshot();
    });
  });
});
