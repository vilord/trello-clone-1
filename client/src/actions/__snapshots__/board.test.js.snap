// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Actions Action Creators creates a SET_BOARD_TITLE action 1`] = `
Object {
  "title": "Some Title",
  "type": "SET_BOARD_TITLE",
}
`;
